### PR TITLE
Enable latest version of qiskit sphinx theme

### PR DESCRIPTION
**Pull request by javabster**
### Summary
The latest version of the qiskit_sphinx_theme (1.10.0) includes 2 major new features:

- It replaces the existing top menu with the same top menu from qiskit.org
- It adds support for segment analytics (optional)

This PR removes the existing `docs/templates/layout.html` so that terra will stop overriding the` qiskit_sphinx_theme` and enable the new top menu to show. This PR also updates the `conf.py` to opt-in to analytics (details on what is tracked [here](https://github.com/Qiskit/qiskit_sphinx_theme#enable-qiskitorg-analytics))

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -76,6 +76,7 @@
 }
 html_static_path = ["_static"]
 html_css_files = []
+html_context = {"analytics_enabled": True}
```
Comment (mtreinish) [inline docs/conf.py:79]: We should only enable analytics in docs jobs when we're publishing to qiskit.org we don't want to include any of this by default. It should be disabled here and for all builds except for those run as part of a publish job.

Comment (javabster) [reply docs/conf.py:79]: What would you suggest? I'm not familiar with the terra build process, but this is the important info about analytics:
- no analytics are collected unless the page is loaded at a qiskit.org domain, so even running locally etc. will not trigger any analytics
- the point of the flag here is to give qiskit projects control over whether they collect analytics data once deployed to qiskit.org. So if this line is removed even if an app is using the latest theme version they will not be collecting analytics data by default even once deployed to qiskit.org

Comment (mtreinish) [reply docs/conf.py:79]: This conf.py isn't used at all for published documentation so we should not set it at all in the terra repo. It should only be set in qiskit/qiskit, I also would only set it via an env variable that gets set during the deployment jobs.

Comment (javabster) [reply docs/conf.py:79]: ah gotcha, I didn't realise the `conf.py` here wasn't even used

**Comment by javabster** [comment]
closing in favour of https://github.com/Qiskit/qiskit/pull/1631

**Pull request closed without merging**